cmake/libigl.cmake

```diff
@@ -80,6 +80,10 @@ if(MSVC)
   # Enable parallel compilation for Visual Studio
   target_compile_options(igl_common INTERFACE /MP /bigobj)
   target_compile_definitions(igl_common INTERFACE -DNOMINMAX)
+  target_compile_options(igl_common INTERFACE /Zc:__cplusplus)
+  if(LIBIGL_USE_STATIC_LIBRARY)
+    target_compile_definitions(igl_common INTERFACE -D_USE_MATH_DEFINES)
```

Comment:
This shouldn't be necessary, as we take care to not use `M_PI` directly in libigl. Is there any reason to enable this flag?

Reply:
Same issue, perhaps it was fixed earlier but in my experiment `M_PI` had crept in somehow. I can remove this too.

Reply:
Yes, we should instead remove all usage of `M_PI` from libigl's code instead, without requiring additional compilation flags from the user.